Dockerfile

```diff
@@ -1,34 +1,15 @@
-FROM ocaml/opam:debian-12-ocaml-4.14@sha256:06d58a5cb1ab7875e8d94848102be43f6492e95320e8e9a9ecb9167654d0ee3f AS build
-RUN sudo ln -f /usr/bin/opam-2.1 /usr/bin/opam && opam init --reinit -ni
+FROM ocaml/opam:debian-12-ocaml-4.14 AS build
```

Comment:
Will is it possible we'll be inviting more instability by unpinning the base image? Or do you think that is not a concern?

An alternative is that we rely on published packages, but still pin our repository down via the image hash, requiring we update this as needed, but insuring packages updates don't break things out from under us... but then, maybe would catch that in our deployability checks (at least 99%)?

Reply:
We pin the base image SHA, and then checkout a specific opam-repository SHA.  Picking the base image SHA seems redundant as the latest image always includes the history.  By picking the latest image we reduce the work needed by `apt update`.  Therefore, I think the discussion should be on whether we fix the opam-repository SHA (line 4).  I took both out in the PR to raise the discussion.  As you say, we have a decision on whether we decide to move this forward each time we publish new packages on opam-repository, or just accept the current version.  On reflection, I think we should fix the opam-repository SHA as this will give us a reliably deployable package.  OCaml-CI will test with the latest opam-repository so we will get advanced warning on issues.  The PR to update the SHA will be the trigger to cause `ocurrent-deployer` to build and deploy the new version.

Reply:
Your proposed way forward there makes sense to me. :+1: